### PR TITLE
Pull request - data loader

### PR DESCRIPTION
**Pull request by lhd231**
Data loader for text files

### CODE REVIEW

**Comment by ghost** [comment]
I looked at Noah's code. It looks good, the tfrecords file is storing the pairs of samples and labels. The code doesn't acquire the data on its own, but I am not sure if this is a necessity. 
Also for both our data, we might want to add a code for accessing the tfrecords file.

File: data/labels_sample.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+1.000000000000000000e+00
```
Comment (rdevon) [inline data/labels_sample.txt:1]: So why do we need this file? It seems like it can be generated on its own. In addition, why are they floats if they are labels?

File: data/snp.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+command line arguments accept at most, 2 arguments.  The file names for the data and labels file
+'''
+
+'''
+numpy is the most effective way, in my experience, to read in numeric data.  Tensorflow does have
+a 'TextLineReader()' which allows for text reading as strings.  This would be useful if the data
+was in byte form (or text).  However, numpy is the most streamlined for numeric data
+'''
+def load_file(name,delimiter=',',type=np.float32):
```
Comment (rdevon) [inline data/snp.py]: You need to delint your code. I will add lint functionality to this repo soon.

File: data/snp.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import tensorflow as tf
+import numpy as np
+import sys
+
+'''
```
Comment (rdevon) [inline data/snp.py]: Please place the descriptions of the script before imports.

File: data/snp.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    return tf.train.Feature(float_list = tf.train.FloatList(value=data))
+
+#default file names
+data_file = "./data_sample.txt"
+labels_file = "./labels_sample.txt"
+
+if __name__ == "__main__":
+    args = sys.argv
+
+    if len(args) == 2:
```
Comment (rdevon) [inline data/snp.py]: I suggest using argparse to deal with arguments. The record is just going to go into the current directory, which is bad practice. For now, include an argument for the output directory / file. I suggest though eventually we have a default "home" for data that is user-defined.

File: data/snp.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        data_file = args[0]
+        labels_file = args[0]
+
+    record_name = "./text_records.tfrecords"
+    features = load_file(data_file)
+    labels = load_file(labels_file)
+    record_writer = tf.python_io.TFRecordWriter(record_name)
+
```
Comment (rdevon) [inline data/snp.py]: Is there any check to make sure the features / labels are the right format?

File: data/snp.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+'''
+def load_file(name,delimiter=',',type=np.float32):
+    return np.loadtxt(name,delimiter=delimiter,dtype=type)
+
+
+def build_records(data):
+    return tf.train.Feature(float_list = tf.train.FloatList(value=data))
+
+#default file names
+data_file = "./data_sample.txt"
```
Comment (rdevon) [inline data/snp.py:34]: I see the need for testing data, but I'm not sure it belongs here. For one, make a unit test that runs on this data. Second, if it's really necessary for testing, can it just be generated (using a random number generator), rather than take up space in the repo. Finally, I suggest using a temporal directory (such as in `/tmp`) to store test data after its generated (and then erase it).

File: data/snp.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        #NOTE: The value argument must be iterable.  In this case, l is a single float.  Which would
+        #not be the case if the data was one_hot
+        labels_record = build_records([l])
+
+        #each feature can be of different sizes.  However, in this case, that is not required
+        output = tf.train.Example(features=tf.train.Features(
+            feature={'data': data_record, 'labels': labels_record}))
+        record_writer.write(output.SerializeToString())
+
+    print "Save completed.  File is: ./text_records.tfrecords"
```
Comment (rdevon) [inline data/snp.py]: We need to move to logging. 

@lhd231 @srinjaypaul @edamaraju I had logging in cortex. Can we just import that functionality here and now?

**Review by rdevon — CHANGES_REQUESTED** [review]

**Comment by rdevon** [comment]
Out of date. DataLoader has changed quite a bit. Any chance this will be updated?
